x-pack/plugins/observability_solution/apm/public/context/entity_manager_context/entity_manager_context.tsx

```diff
@@ -23,15 +36,34 @@ export function EntityManagerEnablementContextProvider({
 }: {
   children: React.ReactChild;
 }) {
-  const { isEnabled, status, refetch } = useEntityManager();
+  const { core } = useApmPluginContext();
+  const { isEnabled: isEntityManagerEnabled, status, refetch } = useEntityManager();
+
+  const [userServiceInventoryView, setUserServiceInventoryView] = useLocalStorage(
+    serviceInventoryStorageKey,
+    ServiceInventoryView.classic
+  );
+
+  const isEntityCentricExperienceSettingEnabled = core.uiSettings.get<boolean>(
+    entityCentricExperience,
+    false
+  );
+
+  const isEntityCentricExperienceViewEnabled =
+    isEntityManagerEnabled &&
+    userServiceInventoryView === ServiceInventoryView.entity &&
+    isEntityCentricExperienceSettingEnabled;
 
   return (
     <EntityManagerEnablementContext.Provider
       value={{
-        isEntityManagerEnabled: isEnabled,
+        isEntityManagerEnabled,
         entityManagerEnablementStatus: status,
         isEnablementPending: status === ENTITY_FETCH_STATUS.LOADING,
         refetch,
+        userServiceInventoryView,
+        setUserServiceInventoryView,
```

Comment:
I'm getting confused with the `user` terminology here. Especially because we don't have per-user settings. Could you try finding a better name that mentions that this is saved in the LS?

Reply:
what about? 
`userServiceInventoryView` -> `serviceInventoryViewLocalStorageSetting`

Reply:
https://github.com/elastic/kibana/pull/188343/commits/a1fad6d627e0b1624072eaeb1efd9caa1fe14651